src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -102,17 +99,18 @@ class BlockingProcess(
 
   private val process = startProcess()
 
+  // This shutdown hook is required because the stdio actors will be blocked on a read. The only
+  // reliable means of unblocking them is to destroy the thing they're listening to.
+  context.actorOf(ShutdownHook.props(() => process.destroy()), "process-destroyer")
```

Comment:
Good question. It is a thread safe as `close` I guess (which we were using). I've submitted a bug report to Oracle in order for them to clarify the thread safety of the call to `destroy`.

That said, I think that the actor shutdown will occur in an orderly manner such that the process object is handled reasonably i.e. the parent actor will do nothing with the process object until its children have shutdown.

Any other thoughts or inspiration would be welcomed. Unfortunately the Java APIs do not provide a PID otherwise we could just kill it via the command line.

I'd actually like to see a new implementation of our Akka `BlockingProcess` api which would be non-blocking and thus perhaps called just `Process`.